jenkins/spark-premerge-build.sh

```diff
@@ -206,7 +206,7 @@ ci_scala213() {
     cd .. # Run integration tests in the project root dir to leverage test cases and resource files
     export TEST_TAGS="not premerge_ci_1"
     export TEST_TYPE="pre-commit"
-    export TEST_PARALLEL=5
```

Comment:
@liurenjie1024  Can you explain why we need to change PARALLEL to 4 from 5?


Please note : this change may make our pre-merge build become slower

Reply:
> This is to workaround this bug: https://github.com/NVIDIA/spark-rapids/issues/8652

Thanks for your info @liurenjie1024 

I did not found connections between this PR and issue/8652, can you help explain it, thanks a lot!

BTW, if the change here `TEST_PARALLEL=4` is a workaround instead fixing, can you help add notation like `//TODOL revert the line change after issue 652 get fixed` ahead of the changing line? 

Please note : this change is critical, it makes our the pre-merge CI pipeline become slower.
It will impact all the coming pre-merge CI if this PR get merged!